tests/compiler/test_strings.py

```diff
@@ -50,3 +50,88 @@ def test_rstring(snippetcompiler):
         """,
     )
     compiler.do_compile()
+
+
+def test_no_multiline_in_single_quoted_string_double_apostrophe(snippetcompiler):
+    snippetcompiler.setup_for_error(
+        """
+std::print(
+  "test
+  string"
+)
+        """,
+        "Syntax error: Illegal character '\"' ({dir}/main.cf:3:3)",
```

Comment:
You're right that this is very unintuitive (many of our syntax errors are). A possible improvement could be to raise the error in the parser rather than the lexer. It's only a start but I think I'm in favor of taking that approach. @wouterdb wdyt?

Reply:
It would improve the error reporting in this case, it might make it worse in case you really have a missing '\"'
(i.e. what would this do)

```
entity Test:
   string blah = "the value
end

[rest of model]

first_string_after_that = "a"
```

or worse

```
entity Test:
   string blah = "the value
end

[rest of model]
entity Test2:
   string blah = the value"
end

```

Reply:
Fair point. To improve that we would need to shift quite a bit more to the parser. Let's leave it as is then?